ksql-engine/src/main/java/io/confluent/ksql/KsqlContext.java

```diff
@@ -41,18 +43,28 @@
   private static final String KAFKA_BOOTSTRAP_SERVER_OPTION_DEFAULT = "localhost:9092";
 
   public static KsqlContext create(final KsqlConfig ksqlConfig) {
-    return create(ksqlConfig, null);
+    return create(
+        ksqlConfig,
+        new KsqlSchemaRegistryClientFactory(ksqlConfig),
+        new KsqlSchemaRegistryClientFactory(ksqlConfig).get());
   }
 
   public static KsqlContext create(
       final KsqlConfig ksqlConfig,
+      final Supplier<SchemaRegistryClient> schemaRegistryClientFactory,
```

Comment:
We do we have a factory and also have an actual instance here? Seems like a confusing interface and the duplication seems to be dropped in downstream calls.